llama_stack/distribution/routers/routers.py

```diff
@@ -242,7 +252,74 @@ async def score(
             ).score(
                 input_rows=input_rows,
                 scoring_functions=[fn_identifier],
+                scoring_params=scoring_params,
             )
             res.update(score_response.results)
 
         return ScoreResponse(results=res)
+
+
+class EvalRouter(Eval):
+    def __init__(
+        self,
+        routing_table: RoutingTable,
+    ) -> None:
+        self.routing_table = routing_table
+
+    async def initialize(self) -> None:
+        pass
+
+    async def shutdown(self) -> None:
+        pass
+
+    async def run_benchmark_eval(
+        self,
+        benchmark_id: str,
+        eval_task_config: BenchmarkEvalTaskConfig,
+    ) -> Job:
+        pass
+
+    async def run_eval(
+        self,
+        eval_task_def: EvalTaskDef,
+        eval_task_config: EvalTaskConfig,
+    ) -> Job:
+        # NOTE: We need to use DEFAULT_EVAL_TASK_IDENTIFIER to make the router work for all app evals
+        return await self.routing_table.get_provider_impl(
+            DEFAULT_EVAL_TASK_IDENTIFIER
+        ).run_eval(
+            eval_task_def=eval_task_def,
+            eval_task_config=eval_task_config,
+        )
+
+    @webmethod(route="/eval/evaluate_rows", method="POST")
+    async def evaluate_rows(
+        self,
+        input_rows: List[Dict[str, Any]],
+        scoring_functions: List[str],
+        eval_task_config: EvalTaskConfig,  # type: ignore
+    ) -> EvaluateResponse:
+        # NOTE: This is to deal with the case where we do not pre-register an eval benchmark_task
+        # We use default DEFAULT_EVAL_TASK_IDENTIFIER as identifier
+        return await self.routing_table.get_provider_impl(
+            DEFAULT_EVAL_TASK_IDENTIFIER
+        ).evaluate_rows(
+            input_rows=input_rows,
+            scoring_functions=scoring_functions,
+            eval_task_config=eval_task_config,
+        )
+
+    async def job_status(self, job_id: str) -> Optional[JobStatus]:
+        return await self.routing_table.get_provider_impl(
+            DEFAULT_EVAL_TASK_IDENTIFIER
```

Comment:
this seems tricky -- couldn't a job be associated with a benchmark also?

Reply:
Currently, we have the eval provider responsible for keeping track of the jobs. Each `job_id` can be benchmark or app eval, and it will be looked up accordingly in the eval provider.

Reply:
Synced offline, we will have a benchmark_id="DEFAULT_APP" for each /eval API.